touchforms/backend/xformplayer.py

```diff
@@ -209,17 +209,12 @@ def walk(self):
 
     def _walk(self, parent_ix, siblings):
         def step(ix, descend):
-            next_ix = self.fem.incrementIndex(ix, descend)
+            next_ix = self.fec.getAdjacentIndex(ix, True, descend)
             self.fem.setQuestionIndex(next_ix)  # needed to trigger events in form engine
             return next_ix
 
         def ix_in_scope(form_ix):
-            if form_ix.isEndOfFormIndex():
-                return False
-            elif parent_ix.isBeginningOfFormIndex():
-                return True
-            else:
-                return FormIndex.isSubElement(parent_ix, form_ix)
+            return form_ix.isInForm()
```

Comment:
[2/10/15, 11:23:51 PM] William Pride: @ your comment, that’s true but those functions seem to be used the same
[2/10/15, 11:25:00 PM] Clayton Sims: hmmm, might be worth seeing if Simon has anything to say, I'm not 100% sure but the beginning index might need to be in scope for certain circumstances?

@snopoke you can see here

https://github.com/dimagi/javarosa/blob/239e578a3b1049aaddf3f9a03612082ebedf92ef/core/src/org/javarosa/form/api/FormEntryController.java#L261

That we've switched to this isInForm() method for basically the same pattern here. The main semantic difference seems to be that the old method returns true when the parent is the beginning of the form - any particular reason to keep this around? (this wasn't necessary for the fix, but I included to update the pattern)